### ISSUE TITLE
Add a login button on public pages

### ISSUE THREAD

**Issue by StCyr** [nextcloud/social#784]
**Describe the solution you'd like**
Add a login button on public pages

**Comment by daita** [comment]
@nextcloud/designers any idea where/how to put a Login button on the public page ? Could it be a core feature to display this button in the top line ?

![Selection_115](https://user-images.githubusercontent.com/1038617/66204618-62e06e80-e692-11e9-9095-cce5d731ff12.png)

**Comment by violoncelloCH** [comment]
I think it would make sense to put this at the right side of the top bar (like the 'download all' button on link shares)

**Comment by jancborchardt** [comment]
Yeah, agree with @violoncelloCH here, top right "Log in" makes sense.

Also, when you press "Follow" it will prompt you to log in, right?

**Comment by StCyr** [comment]
> 
> 
> Yeah, agree with @violoncelloCH here, top right "Log in" makes sense.
> 
> Also, when you press "Follow" it will prompt you to log in, right?

@daita's example is a bit misleading: Currently, the profile component isn't shown on every public page. Hence, the proposal.

Here's an example of a public page without a follow button:

![image](https://user-images.githubusercontent.com/8179031/66292274-36239580-e8e4-11e9-93ae-c76794240d96.png)

But, maybe the correct solution is to add a profile component on this page?

**Comment by SchBenedikt** [comment]
> Note: This comment was created by Opencode with Muse Spark 1.3 Free (AI-assisted triage). Please correct me if this assessment is wrong.

A login entry point for visitors on public pages now exists (PR #2279, commit `7c630b81` "feat(public): give a visitor a way into their own account"), so I am closing this:

- `src/App.vue:29-36` renders a visitor banner whenever `serverData.public` is true ("You are reading this as a visitor." + primary button `Log in to {host}`).
- The target is computed in `src/App.vue:206-220` (`loginUrl()` → `/login?redirect_url=...`, preserving the public page for redirect after login).
- It is wired for all public surfaces via `lib/Controller/SocialPubController.php:118-120` (`renderPage()`: profile, followers, following, collections, portfolio) and `:246-250` (`displayPost()`), which set `serverData.public=true`.
- Covered by `tests/js/App.test.js` ("a visitor reading a public page").

Known limitation (out of scope here): the guest 404 page (`templates/notfound.php:15-24`) only offers "Back to …", no login. Reopen if the banner does not appear for you on a public page.
